src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -705,102 +673,95 @@ def _on_current_coordinates_changes(self, coordinates: ListVA) -> None:
+                # Set x and y position in the grid
+                self.grid.SetCellValue(current_row_count, GridColumns.X.value,
+                                       f"{pixel_coords[0]:.{GRID_PRECISION}f}")
+                self.grid.SetCellValue(current_row_count, GridColumns.Y.value,
+                                       f"{pixel_coords[1]:.{GRID_PRECISION}f}")
+                self.grid.SetCellValue(current_row_count, GridColumns.Index.value, str(target.index.value))
+                self.grid.SetCellValue(current_row_count, GridColumns.Type.value, target.name.value)
 
         self._reorder_grid()
         self._panel.Layout()
-        self._update_feature_correlation_target()
+        self.correlation_target = update_feature_correlation_target(self.correlation_target, self._tab_data_model)
 
         for vp in self._viewports:
             vp.canvas.update_drawing()
 
         if self.check_correlation_conditions():
             self._need_reprocessing()
 
-    def _on_z_targeting(self, event) -> None:
+    def _on_z_targeting(self, evt) -> None:
         """
         Handle Z-targeting when the Z-targeting button is clicked.
         """
-        # TODO change the color based on success in the grid
         if self._tab_data_model.main.currentTarget.value:
+
             # Select the streams which are visible in the view for Z-targeting
             streams_projections = self._tab_data_model.views.value[0].stream_tree.flat.value
             if not streams_projections:
                 wx.MessageBox("FM streams are not available for refining Z", "Error", wx.OK | wx.ICON_ERROR)
                 return
-            das = [stream_projection.stream.raw[0] for stream_projection in streams_projections]
+
             coords = self._tab_data_model.main.currentTarget.value.coordinates.value
-            pixel_coords = self.correlation_target.fm_streams[0].getPixelCoordinates((coords[0], coords[1]),
-                                                                                     check_bbox=False)
-            self._tab_data_model.main.currentTarget.value.coordinates.value[2] = float(get_optimized_z_gauss(das, int(
-                pixel_coords[0]), int(pixel_coords[1]), coords[2]))
+            pixel_coords = self.correlation_target.fm_streams[0].getPixel3DCoordinates(coords)
+            das = [interpolate_z_stack(da=stream_projection.stream.raw[0][:,int(pixel_coords[1]):int(pixel_coords[1])+1,
+                   int(pixel_coords[0]):int(pixel_coords[0])+1], method="linear")
+                   for stream_projection in streams_projections]
+
+            z = float(get_optimized_z_gauss(das,int(0),int(0),coords[2]))
+            self._tab_data_model.main.currentTarget.value.coordinates.value[2] = self.correlation_target.fm_streams[0].getPhysical3DCoordinates((pixel_coords[0],
+                                                                                pixel_coords[1], z))[2]
```

Comment:
Check the formatting. Space after `,`. Use intermediary variables to fit things on 1 line (eg, `phy_z`).